src/rule/index.ts

```diff
@@ -59,15 +61,60 @@ const incrementHit = (node: { x_x_x_x_results: a.ReportResult }) => {
   node.x_x_x_x_results.hits++;
 };
 
-const removeServer = (api: a.Root, path: string) => {
+// Named sloppy since there are many cases this function does not catch
+const isAbsoluteUrlSloppy = (urlString: string) => {
+  return (
+    urlString.indexOf("http://") === 0 || urlString.indexOf("https://") === 0
+  );
+};
+
+export const removeServer = (
+  api: a.Root,
+  url: URL,
+  url_server_prefix: string
+): string | null => {
   // For now we can only have one server
-  const prefix = api.servers[0].url;
-  if (path.startsWith(prefix)) {
-    const result = path.slice(prefix.length);
-    return result;
-  } else {
-    return path;
+  const server = api.servers[0].url;
+  const pathToMatch = url.pathname;
+
+  if (!isAbsoluteUrlSloppy(server)) {
+    // Open Api server can start with slash, remove to simplify logic
+    let serverToMatch = server;
+    if (!server.startsWith("/")) {
+      serverToMatch = "/" + serverToMatch;
+    }
+
+    // If the relative server ends with a trailing slash we remove to simplify logic
+    if (server.endsWith("/")) {
+      serverToMatch = serverToMatch.slice(0, serverToMatch.length - 2);
+    }
+
+    if (pathToMatch.indexOf(serverToMatch) === 0) {
+      return pathToMatch.slice(serverToMatch.length, pathToMatch.length);
```

Comment:
`"foo".slice(1, 1)` will return `""`. There seems to be no test case where this happen. Can it happen?